Client/Frontend/Widgets/ToolbarButtons.swift

```diff
@@ -0,0 +1,185 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+enum ToolbarButtonTag: Int {
+    case Back
+    case Forward
+    case Reload
+    case Share
+    case Bookmarked
+}
+
+// MARK: - Browser Toolbar Buttons
+extension UIButton {
```

Comment:
I agree - it seems odd to have specific button extensions for 'Toolbar' items. I wonder if I can typealias/associatedtype UIButton -> ToolbarButton and extend off that. That way the scope makes sense and the type we pass into the containers for the URLBarView will be constrained to ToolbarButtons instead of anything.